source/core/Ky.ts

```diff
@@ -217,7 +217,9 @@ export class Ky {
 				throw error;
 			}
 
-			const retryAfter = error.response.headers.get('Retry-After');
+			const retryAfter = error.response.headers.get('Retry-After')
+				?? error.response.headers.get('RateLimit-Reset')
+				?? error.response.headers.get('X-RateLimit-Reset');
```

Comment:
I don’t think this correct. On mdn I see two formats for `Retry-After` an ISO date or a number of seconds to wait

```
Retry-After: Wed, 21 Oct 2015 07:28:00 GMT
Retry-After: 120
``` 

While [GitHub's docs](https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api?apiVersion=2022-11-28#handle-rate-limit-errors-appropriately) say:

> should not make another request until after the time specified by the x-ratelimit-reset header. The x-ratelimit-reset header is in UTC epoch seconds.

So this introduces a third type: UTC epoch. I think this is easy to detect though: `if t is number and t > 1700000000`

Reply:
I used the semantics defined by the IETF draft:
https://www.ietf.org/archive/id/draft-polli-ratelimit-headers-02.html#name-ratelimit-reset

> The header value is
>
> ```
> RateLimit-Reset = delta-seconds
> ```
>
> The delta-seconds format is used because:
> -  it does not rely on clock synchronization and is resilient to clock adjustment and clock skew between client and server (see [[RFC7231](https://www.ietf.org/archive/id/draft-polli-ratelimit-headers-02.html#RFC7231)] Section 4.1.1.1);
> - it mitigates the risk related to thundering herd when too many clients are serviced with the same timestamp.

> An example of RateLimit-Reset use is below.
> ```
> RateLimit-Reset: 50
> ```

Looks like neither X.com nor GitHub actually follow this proposal.

I also just noticed that X.com uses a slightly different header name: `x-rate-limit-reset` (note the `-` between `rate` and `limit`)

Reply:
I found this very useful thread regarding header names and semantics:
https://github.com/ietf-wg-httpapi/ratelimit-headers/issues/25

I'll leave it to others to add some of the more obscure formats, if the need arises.